target/linux/ath79/dts/ar7242_ubnt_edgeswitch-8xp.dts

```diff
@@ -0,0 +1,306 @@
+			gpios = <&gpio 13 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	spi {
+		compatible = "spi-gpio";
+		#address-cells = <0x1>;
+		ranges;
+
+		gpio-sck = <&gpio 16 GPIO_ACTIVE_HIGH>;
+		gpio-mosi = <&gpio 15 GPIO_ACTIVE_HIGH>;
+		cs-gpios = <&gpio 14 GPIO_ACTIVE_HIGH>;
+		num-chipselects = <1>;
+
+		gpio_hc595: gpio_spi@0 {
+			compatible = "fairchild,74hc595";
+			reg = <0>;
+			registers-number = <2>;
+			spi-max-frequency = <100000>;
+			enable-gpios = <&gpio 7 GPIO_ACTIVE_HIGH>;
+
+			gpio-controller;
+			#gpio-cells = <2>;
+
+		};
+	};
+
+
+	gpio_poe: gpio-export {
+		compatible = "gpio-export";
+
+		poe_24v_port1 {
+			gpio-export,name = "ubnt:24v-poe:port1";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 0 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_48v_port1 {
+			gpio-export,name = "ubnt:48v-poe:port1";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 1 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_24v_port2 {
+			gpio-export,name = "ubnt:24v-poe:port2";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 2 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_48v_port2 {
+			gpio-export,name = "ubnt:48v-poe:port2";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 3 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_24v_port3 {
+			gpio-export,name = "ubnt:24v-poe:port3";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 4 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_48v_port3 {
+			gpio-export,name = "ubnt:48v-poe:port3";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 5 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_24v_port4 {
+			gpio-export,name = "ubnt:24v-poe:port4";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 6 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_48v_port4 {
+			gpio-export,name = "ubnt:48v-poe:port4";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 7 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_24v_port5 {
+			gpio-export,name = "ubnt:24v-poe:port5";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 8 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_48v_port5 {
+			gpio-export,name = "ubnt:48v-poe:port5";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 9 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_24v_port6 {
+			gpio-export,name = "ubnt:24v-poe:port6";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 10 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_48v_port6 {
+			gpio-export,name = "ubnt:48v-poe:port6";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 11 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_24v_port7 {
+			gpio-export,name = "ubnt:24v-poe:port7";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 12 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_48v_port7 {
+			gpio-export,name = "ubnt:48v-poe:port7";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 13 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_24v_port8 {
+			gpio-export,name = "ubnt:24v-poe:port8";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 14 GPIO_ACTIVE_HIGH>;
+		};
+
+		poe_48v_port8 {
+			gpio-export,name = "ubnt:48v-poe:port8";
+			gpio-export,output = <0>;
+			gpios = <&gpio_hc595 15 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&spi {
```

Comment:
This overwrites the partitioning in ubnt_sw with a different one.
So, as it does not seem obvious what should be in ubnt_sw.dtsi and we only have on device inheriting from it, please just move all the definitions there to this DTS file and drop the DTSI.

Reply:
Thorught adding the dtsi might be a good idea since all SW devices seem to be the same except for using different switches/switch configurations. But if keeping it all a single dts as long as there is only one supported device is policy I'll stick with that.

Reply:
Partition overwrite was indeed copy&paste foo. Didn't mean to leave that in the dts.